codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/JsonRpcProtocolGenerator.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.aws.go.codegen;
+
+import java.util.Set;
+import software.amazon.smithy.codegen.core.SymbolProvider;
+import software.amazon.smithy.go.codegen.GoStackStepMiddlewareGenerator;
+import software.amazon.smithy.go.codegen.GoWriter;
+import software.amazon.smithy.go.codegen.SmithyGoDependency;
+import software.amazon.smithy.go.codegen.integration.HttpRpcProtocolGenerator;
+import software.amazon.smithy.go.codegen.integration.ProtocolGenerator;
+import software.amazon.smithy.go.codegen.integration.ProtocolUtils;
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.model.shapes.OperationShape;
+import software.amazon.smithy.model.shapes.Shape;
+import software.amazon.smithy.model.shapes.StructureShape;
+
+/**
+ * Handles generating the aws.rest-json protocol for services.
+ *
+ * @inheritDoc
+ *
+ * @see HttpRpcProtocolGenerator
+ */
+abstract class JsonRpcProtocolGenerator extends HttpRpcProtocolGenerator {
+
+    /**
+     * Creates an AWS JSON RPC protocol generator
+     */
+    public JsonRpcProtocolGenerator() {
+        super(true);
+    }
+
+    @Override
+    protected String getOperationPath(GenerationContext context, OperationShape operation) {
+        return "/";
+    }
+
+    @Override
+    protected void writeDefaultHeaders(GenerationContext context, OperationShape operation, GoWriter writer) {
+        super.writeDefaultHeaders(context, operation, writer);
+        // TODO: generate request signer (unsigned body)
+        String target = context.getService().getId().getName() + "." + operation.getId().getName();
```

Comment:
is this TODO still valid?

Reply:
It could stand to be re-worded or resolved, but currently the unsigned payload version of the signer isn't being attached.

Reply:
Ah i think https://github.com/aws/aws-sdk-go-v2/pull/625 will resolved that issue.